types/react-native-auth0/index.d.ts

```diff
@@ -0,0 +1,171 @@
+// Type definitions for react-native 1.2
+// Project: https://github.com/auth0/react-native-auth0
+// Definitions by: Andrea Ascari <https://github.com/ascariandrea>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+/**
+ * Auth
+ */
+
+export interface AuthorizationUrlParams {
+    responseType: string;
+    redirectUri: string;
+    state: string;
+}
+
+export interface CreateUserParams<T extends any> {
+    email: string;
+    username?: string;
+    password: string;
+    connection: string;
+    metadata?: T;
+}
+
+export interface CreateUserResponse {
+    Id: string;
+    emailVerified: boolean;
+    email: string;
+}
+
+export interface ExchangeParams {
+    code: string;
+    redirectUri: string;
+    verifier: string;
+}
+
+export interface LogoutParams {
+    federated: boolean;
+    clientId?: string;
+    returnTo?: string;
+}
+
+export interface PasswordRealmParams {
+    username: string;
+    password: string;
+    realm: string;
+    audience?: string;
+    scope?: string;
+}
+
+export interface PasswordRealmResponse {
+    accessToken: string;
+    expiresIn: number;
+    idToken: string;
+    scope: string;
+    tokenType: "Bearer";
+}
+
+export interface RefreshTokenParams {
+    refreshToken: string;
+    scope?: string;
+}
+
+export interface RevokeParams {
+    refreshToken: string;
+}
+
+export interface UserInfoParams {
+    token: string;
+}
+
+export interface UserInfo {
+    email: string;
+    emailVerified: boolean;
+    name: string;
+    nickname: string;
+    picture: string;
+    sub: string;
+    updatedAt: string;
+}
+export class Auth {
+    authorizationUrl(params: AuthorizationUrlParams): string;
+    createUser(user: CreateUserParams<any>): Promise<CreateUserResponse>;
+    exchange(params: ExchangeParams): Promise<string>;
+    logoutUrl(params: LogoutParams): string;
+    passwordRealm(params: PasswordRealmParams): Promise<PasswordRealmResponse>;
+
+    refreshToken(params: RefreshTokenParams): Promise<any>;
+    revoke(params: RevokeParams): Promise<any>;
+    userInfo(params: UserInfoParams): Promise<UserInfo>;
+}
+
+/**
+ * Users
+ */
+export interface Auth0User<T> {
+    created_at: string;
+    email: string;
+    emailVerified: boolean;
+    identities: any[];
+    last_ip?: string;
+    last_login?: string;
+    logins_count: number;
+    name: string;
+    nickname: string;
+    picture?: string;
+    updated_at: string;
+    userId: string;
+    userMetadata?: T;
+}
+
+export interface GetUserParams {
+    id: string;
+}
+
+export interface PatchUserParams<T> {
+    id: string;
+    metadata: T;
+}
+
+export class Users {
+    constructor(options: UsersOptions);
+    /* tslint:disable-next-line no-unnecessary-generics */
+    getUser<T extends any>(parameters: GetUserParams): Promise<Auth0User<T>>;
```

Comment:
I'm always a bit skeptical of these signatures because it's easy to forget the type parameter and are as effective as type assertions.

Reply:
Are you suggesting to remove the *extends any* and let the user define `T`?